moveit_ros/planning/planning_request_adapter_plugins/src/publish_intermediate_trajectory.cpp

```diff
@@ -0,0 +1,81 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2012, Willow Garage, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Bianca Homberg, based off empty.cpp by Ioan Sucan */
+
+#include <moveit/planning_request_adapter/planning_request_adapter.h>
+#include <class_loader/class_loader.hpp>
+
+namespace default_planner_request_adapters
+{
+class PublishIntermediateTrajectory : public planning_request_adapter::PlanningRequestAdapter
+{
+public:
+  ros::Publisher intermediate_trajectory_;
+  ros::NodeHandle nh_;
+
+  PublishIntermediateTrajectory() : planning_request_adapter::PlanningRequestAdapter(), nh_("~")
+  {
+  }
+
+  void initialize(const ros::NodeHandle& node_handle) override
+  {
+    std::string topic_name = "intermediate_trajectory";
```

Comment:
Talked this over in person with @bhomberg - there doesn't seem to be any way to distinguish between different planning adapters (the planning_pipeline makes a basic assumption that there is never more than one planning adapter of any given type). So for now this seems fine. A future improvement would be to fundamentally change the way planning adapters are loaded to allow multiple of the same type.

Reply:
how about a static variable to count up for different instances?

Reply:
Multi plugins of the same type within the same pipeline would all publish on the same topic. Longer term it would be best to modify the way the pipeline is specified so that each adapter has a name provided by the user, and it can load params and publish on topics based on that uniqe name. I like the static variable workaround though! What do you think, @bhomberg?